appium-tests/android/android.spec.js

```diff
@@ -27,52 +28,47 @@
 
 'use strict';
 
-var wdHelper = require('../helpers/wdHelper');
+var wdHelper = global.WD_HELPER;
+var screenshotHelper = global.SCREENSHOT_HELPER;
 var wd = wdHelper.getWD();
 var cameraConstants = require('../../www/CameraConstants');
 var cameraHelper = require('../helpers/cameraHelper');
-var screenshotHelper = require('../helpers/screenshotHelper');
 
-var STARTING_MESSAGE = 'Ready for action!';
-var RETRY_COUNT = 3; // how many times to retry taking a picture before failing
 var MINUTE = 60 * 1000;
+var BACK_BUTTON = 4;
 var DEFAULT_SCREEN_WIDTH = 360;
 var DEFAULT_SCREEN_HEIGHT = 567;
 var DEFAULT_WEBVIEW_CONTEXT = 'WEBVIEW';
+var PROMISE_PREFIX = 'appium_camera_promise_';
 
 describe('Camera tests Android.', function () {
     var driver;
     // the name of webview context, it will be changed to match needed context if there are named ones:
     var webviewContext = DEFAULT_WEBVIEW_CONTEXT;
     // this indicates that the device library has the test picture:
     var isTestPictureSaved = false;
-    // this indicates that there was a critical error and we should try to recover:
-    var errorFlag = false;
-    // this indicates that we couldn't restore Appium session and should fail fast:
-    var stopFlag = false;
     // we need to know the screen width and height to properly click on an image in the gallery:
     var screenWidth = DEFAULT_SCREEN_WIDTH;
     var screenHeight = DEFAULT_SCREEN_HEIGHT;
+    // promise count to use in promise ID
+    var promiseCount = 0;
 
-    function win() {
-        expect(true).toBe(true);
+    function getNextPromiseId() {
+        return PROMISE_PREFIX + promiseCount++;
```

Comment:
Nit: it would be better to write `PROMISE_PREFIX + (++promiseCount)`, then in `getCurrentPromiseId` you could just return `PROMISE_PREFIX + promiseCount`